integration_test.go

```diff
@@ -118,23 +120,48 @@ var _ = Describe("When creating CRDs sequentially after resources are created",
 			_, err := k8sClient.Resource(gvrStorageAccount).Namespace("default").Create(context.TODO(), &objStorageAccount, metav1.CreateOptions{})
 			Expect(err).To(BeNil())
 		}, 30)
-		It("should create the storage account and assign the status.id", func() {
-			By("returning the storage account ID")
-			Eventually(func() string {
+		It("should create the storage account and assign the status", func() {
+			By("returning the storage account properties in the status")
+			Eventually(func() bool {
 				obj, err := k8sClient.Resource(gvrStorageAccount).Namespace("default").Get(context.TODO(), storageAccountName, metav1.GetOptions{})
 				Expect(err).To(BeNil())
 
 				status, ok := obj.Object["status"].(map[string]interface{})
 				Expect(err).To(BeNil())
 				if !ok {
-					return ""
+					return false
 				}
 
 				id := status["id"].(string)
-				return id
-			}, time.Minute*3, time.Second*5).Should(Not(BeEmpty())) // TODO check id format
+				Expect(id).Should(MatchRegexp("/subscriptions/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/resourceGroups/" + resourceGroupName))
+				name := status["name"].(string)
+				Expect(name).Should(Not(BeEmpty()))
+				primaryAccessKey := status["primary_access_key"].(string)
+				Expect(primaryAccessKey).Should(Not(BeEmpty()))
+				return true
+			}, time.Minute*3, time.Second*5).Should(BeTrue())
+		}, 300)
+		It("should create the storage account and assign the status with sensitive values", func() {
+			By("returning the storage account primary_access_key encrypted")
+			Eventually(func() error {
+				if encryptionKey, exists := os.LookupEnv("TF_STATE_ENCRYPTION_KEY"); !exists || encryptionKey == "" {
+					Skip("TF_STATE_ENCRYPTION_KEY not set!")
+				}
+
+				obj, err := k8sClient.Resource(gvrStorageAccount).Namespace("default").Get(context.TODO(), storageAccountName, metav1.GetOptions{})
+				Expect(err).To(BeNil())
+
+				status, ok := obj.Object["status"].(map[string]interface{})
+				Expect(err).To(BeNil())
+				if !ok {
+					return fmt.Errorf("status not found in object map")
+				}
 
-			// TODO - check other `status` properties when we're mapping them
+				// If encoded, assume encrypted
+				primaryAccessKey := status["primary_access_key"].(string)
+				_, err = base64.StdEncoding.DecodeString(primaryAccessKey)
```

Comment:
Read this before the reading the comment and was like, um base64 isn't encryption ... now understand.

Reply:
😂 yh we could also run a regex to check this isn't just an encoded connection string but seemed overkill for the time being